llvm/lib/Target/Xtensa/XtensaISelLowering.cpp

```diff
@@ -665,12 +693,30 @@ SDValue XtensaTargetLowering::getAddrPCRel(SDValue Op,
 SDValue XtensaTargetLowering::LowerConstantPool(ConstantPoolSDNode *CP,
                                                 SelectionDAG &DAG) const {
   EVT PtrVT = getPointerTy(DAG.getDataLayout());
```

Comment:
Always reuse the type of the value you always have, you should practically never need to call getPointerTy